src/contracts/wrappers/ERC1155FloorWrapper.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.4;
+
+import {IERC165} from "@0xsequence/erc-1155/contracts/interfaces/IERC165.sol";
+import {IERC1155} from "@0xsequence/erc-1155/contracts/interfaces/IERC1155.sol";
+import {IERC1155TokenReceiver} from "@0xsequence/erc-1155/contracts/interfaces/IERC1155TokenReceiver.sol";
+import {ERC1155} from "@0xsequence/erc-1155/contracts/tokens/ERC1155/ERC1155.sol";
+import {ERC1155MintBurn} from "@0xsequence/erc-1155/contracts/tokens/ERC1155/ERC1155MintBurn.sol";
+import {IERC1155FloorWrapper} from "../interfaces/IERC1155FloorWrapper.sol";
+import {IERC1155Metadata} from "../interfaces/IERC1155Metadata.sol";
+import {IDelegatedERC1155Metadata} from "../interfaces/IDelegatedERC1155Metadata.sol";
+import {WrapperErrors} from "../utils/WrapperErrors.sol";
+
+/**
+ * @notice Allows users to wrap any amount of a ERC-1155 token with a 1:1 ratio.
+ * Therefore each ERC-1155 within a collection can be treated as if fungible.
+ */
+contract ERC1155FloorWrapper is
+    IERC1155FloorWrapper,
+    ERC1155MintBurn,
+    IERC1155Metadata,
+    IERC1155TokenReceiver,
+    WrapperErrors
+{
+    bool private initialized;
+    address internal immutable factory;
+    IERC1155 public token;
+    // This contract only supports a single token id
+    uint256 public constant TOKEN_ID = 0;
+
+    bool private isDepositing;
+
+    constructor() {
+        factory = msg.sender;
+    }
+
+    function initialize(address tokenAddr) external {
+        if (initialized || msg.sender != factory) {
```

Comment:
I know this is for being on the "safe side" but I don't think `initialized` is needed here. With `msg.sender` + `token != address(0)` should be enough.

Reply:
How about going further and only relying on `msg.sender != factory`? Will be safe since the factory will only ever call this function once immediately after creation.

Reply:
`token != address(0)` has practically no additional cost (since the store has to be written too anyway), you are right that `msg.sender != factory` should be enough, but having an extra "correctness" check doesn't hurt in this case.